unstructured/ingest/interfaces.py

```
@@ -246,35 +251,38 @@ class ChunkingConfig(BaseConfig):
     overlap_all: t.Optional[bool] = None
 
     def chunk(self, elements: t.List[Element]) -> t.List[Element]:
-        chunking_strategy = (
-            self.chunking_strategy
-            if self.chunking_strategy in ("basic", "by_title")
-            else "by_title" if self.chunk_elements is True else None
-        )
-
-        if chunking_strategy == "by_title":
-            return chunk_by_title(
-                elements=elements,
-                combine_text_under_n_chars=self.combine_text_under_n_chars,
-                include_orig_elements=self.include_orig_elements,
-                max_characters=self.max_characters,
-                multipage_sections=self.multipage_sections,
-                new_after_n_chars=self.new_after_n_chars,
-                overlap=self.overlap,
-                overlap_all=self.overlap_all,
-            )
-
-        if chunking_strategy == "basic":
-            return chunk_elements(
-                elements=elements,
-                include_orig_elements=self.include_orig_elements,
-                max_characters=self.max_characters,
-                new_after_n_chars=self.new_after_n_chars,
-                overlap=self.overlap,
-                overlap_all=self.overlap_all,
-            )
+        # -- chunk_elements is a legacy cli argument used for "by_title" chunking that we need to
+        # -- keep supporting.
+        if self.chunk_elements is True:
+            if self.chunking_strategy is not None:
+                logger.warning(
+                    "Both chunk_elements and chunking_strategy were defined."
+                    "chunk_elements is deprecated in favor of chunking_strategy."
+                    f"Defaulting to chunking_strategy: {self.chunking_strategy}."
+                )
```

this should be a failure instead of a warning

---

The warning (and validation in general) should be raised at the CLI, though